src/Microsoft.Extensions.Logging/LoggerExternalScopeProvider.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.Logging
+{
+    public class LoggerExternalScopeProvider : IExternalScopeProvider
+    {
+        private readonly AsyncLocal<Scope> _currentScope = new AsyncLocal<Scope>();
+
+        public void CollectScope<T>(Action<object, T> callback, T state)
```

Comment:
Put a doc comment on here describing how the walk is done (from leaf nodes up to the root). Doesn't have to be amazing prose, but something that makes the behavior relatively clear for us to wordsmith later (and to help provider authors)